packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -148,21 +147,22 @@ export async function executePlugins(
 
   logMultipleResults(results, 'Plugins', undefined, errorsCallback);
 
-  // TODO: add groupBy method for promiseSettledResult by status, #287
-
-  const failedResults = results.filter(isPromiseRejectedResult);
-  if (failedResults.length) {
-    const errorMessages = failedResults
+  const { fulfilled, rejected } = groupByStatus(results) as {
+    fulfilled: PromiseFulfilledResult<PluginReport>[];
+    rejected: PromiseRejectedResult[];
+  };
```

Comment:
The type-cast is unnecessary and could mask potential errors:

```suggestion
  const { fulfilled, rejected } = groupByStatus(results);
```